Client/Frontend/Library/HistoryPanel/HistoryPanel.swift

```diff
@@ -0,0 +1,651 @@
+        
+        return siteItem
+    }
+
+    private func showClearRecentHistory() {
+        clearHistoryHelper.showClearRecentHistory(onViewController: self, didComplete: { [weak self] date in
+            if let date = date {
+                self?.viewModel.removeVisibleSectionFor(date: date)
+            } else {
+                // The only time there's no date, is when we are deleting everything.
+                self?.viewModel.visibleSections = []
+            }
+            
+            self?.viewModel.reloadData()
+            self?.applySnapshot(animatingDifferences: true)
+            
+            if let cell = self?.clearHistoryCell {
+                self?.setTappableStateAndStyle(with: HistoryActionablesModel.activeActionables.first(where: { $0.itemIdentity == .clearHistory }), on: cell)
+            }
+        })
+    }
+    
+    func siteForIndexPath(_ indexPath: IndexPath) -> Site? {
+        // First section is reserved for fixed history actionables.
+        guard indexPath.section > HistoryPanelSections.additionalHistoryActions.rawValue else {
+            return nil
+        }
+
+        let sitesInSection = viewModel.groupedSites.itemsForSection(indexPath.section - 1)
+        return sitesInSection[safe: indexPath.row]
+    }
+
+    func handleNotification(_ notification: Notification) {
+        switch notification.name {
+        case .FirefoxAccountChanged, .PrivateDataClearedHistory:
+            viewModel.groupedSites = DateGroupedTableData<Site>()
+            reloadData()
+
+            if profile.hasSyncableAccount() {
+                resyncHistory()
+            }
+            break
+        case .DynamicFontChanged:
+            reloadData()
+
+            if emptyStateOverlayView.superview != nil {
+                emptyStateOverlayView.removeFromSuperview()
+            }
+            emptyStateOverlayView = createEmptyStateOverlayView()
+            resyncHistory()
+            break
+        case .DatabaseWasReopened:
+            if let dbName = notification.object as? String, dbName == "browser.db" {
+                reloadData()
+            }
+        case .OpenClearRecentHistory:
+            showClearRecentHistory()
+        default:
+            // no need to do anything at all
+            browserLog.error("Error: Received unexpected notification \(notification.name)")
+            break
+        }
+    }
+
+    // MARK: - UITableViewDataSource
+    
+    /// Handles dequeuing the appropriate type of cell when needed.
+    private func configureDatasource() {
+        diffableDatasource = UITableViewDiffableDataSource<HistoryPanelSections, AnyHashable>(tableView: tableView) { [weak self] (tableView, indexPath, item) -> UITableViewCell? in
+            guard let self = self else {
+                os_log(.error, log: .default, "History Panel - self became nil inside diffableDatasource!")
```

Comment:
I believe without access to self, we can't use browser log. So here, I log via os_log.

Reply:
Due to the Loggable protocol, but otherwise self isn't necessary.
And also, other question, is that something we wanna log? History Panel will get deinit at one point.

Reply:
And also, are we retrieving anywhere the os_log? It's something on it's own and unless specifically retrieved, we'll never use it. If we don't have a process in place to review meaningful logs (like we have with Sentry), then not sure it's worth using this.